src/training/scheduler.h

```diff
@@ -175,7 +184,13 @@ class Scheduler : public TrainingObserver {
   }
 
   void started() { LOG(info, "Training started"); }
-  void finished() { LOG(info, "Training finished"); }
+  void finished() {
+    if (keepGoing(false)) // false means: ignore sigterm flag
```

Comment:
Is this the only use of `keepGoing(false)`, i.e. only for logging a different message? If so, I would suggest removing the argument from `keepGoing()` (and keeping it simple) and adding here `if(sigterm_)` or similar.

Reply:
It is possible that both are true: SIGTERM recevied and training finished. In that case, we want to log "Training finished.", not "Training interrupted.", so we'd have to say
```
if(sigterm&&!keepGoing()){
...
```
And that's the opposite of "keeping things simple".

Reply:
OK, but `keepGoing()` is a part of the public interface of the class, so I would value its simpler version more than a simpler if-statement hidden in a function (plus a comment). It also seems the argument is used only internally inside the class.

Reply:
I think @snukky is right here. If that's only for logging purposes just checking `sigterm_` seems fine.